src/Crowdsale/DynamicPriceSale.sol

```diff
@@ -134,18 +137,20 @@ contract DynamicSale is ContextUpgradeable, ReentrancyGuardUpgradeable, Ownable2
     /// @return totalCost TODO
     function calculateTotalCost(uint256 amount) public view returns (uint256 newPrice, uint256 totalCost) {
         uint256 currentSupply = token.totalSupply();
-        require(currentSupply >= priceCurveMinValue, "DynamicSale: current totalSupply too low");
-        require(currentSupply + amount <= priceCurveMaxValue, "DynamicSale: totalSupply limit reached");
-        /// @dev sale-function coefficients
-        uint256 c = 420;
-        uint256 b = 32000461777723 * (10**54);
-        uint256 a = 11680057722 * (10**36);
-
-        uint256 start = currentSupply;
-        uint256 end = start + amount;
-        newPrice = c - a / end**2 + b / end**3;
-        totalCost = c * (end - start) + a * (1 / end - 1 / start) - (b / 2) * (1 / end**2 - 1 / start**2);
+        // require(currentSupply >= priceCurveMinValue, "DynamicSale: current totalSupply too low");
+        // require(currentSupply + amount <= priceCurveMaxValue, "DynamicSale: totalSupply limit reached");
+        // /// @dev sale-function coefficients
+
+        int256 c = 420;
```

Comment:
Why use `int256`? These numbers can never be negative, no?

Reply:
They can't, but they will need to be operated with negative numbers so they need to be casted, otherwise the compiler will not allow it.